aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -105,4 +105,9 @@ public void initialize(Map<String, String> properties) {
     this.awsClientFactory = AwsClientFactories.from(properties);
     this.s3 = awsClientFactory::s3;
   }
+
+  @Override
+  public void close() {
+    client().close();
```

Comment:
Since we'e not 100% sure if `close` will be called more than once, should we set `client` to `null` or add an `AtomicBoolean closed` that will then handle the idempotency issue?